ios/Handlers/RegistrationHandler.swift

```diff
@@ -89,6 +100,39 @@ extension RegistrationHandler : RegistrationConnectorToHandlerProtocol {
 
     func cancelRegistration() {
         handleRedirectURL(url: nil)
+        guard let createPinChallenge = self.createPinChallenge else { return }
+        createPinChallenge.sender.cancel(createPinChallenge)
+    }
+    
+    func handlePinAction(_ pin: String?, action: PinAction) {
+        switch action {
+            case PinAction.provide:
```

Comment:
as now we have typed action, you can ommit `PinAction` in cases and keep just `.provide`, `.cancel`